### PR TITLE
Volume: Avoid unnecessary error trace about limiting to VOL_MIN

### PR DESCRIPTION
**Pull request by singalsu**
The trace issue is only cosmetic since VOL_MIN is currently defined
to zero. There linear gain value was not altered despite the emitted
trace. With this change the limiting no more happens for input value
zero.

The related trace messages are improved to show the defined limits
for easier adjust the offending topology volume scale if this
happens. The string split style is also updated to current.

Signed-off-by: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>

### CODE REVIEW

**Comment by singalsu** [comment]
The SOF issue is #2271.

File: src/audio/volume/volume.c [hunk 1]
```diff
@@ -276,7 +276,7 @@ static int volume_params(struct comp_dev *dev,
  * \param[in] chan Channel number.
  * \param[in] vol Target volume.
  */
-static inline int volume_set_chan(struct comp_dev *dev, int chan, uint32_t vol)
+static inline int volume_set_chan(struct comp_dev *dev, int chan, int32_t vol)
 {
```
Comment (dbaluta) [inline src/audio/volume/volume.c]: not clear why are you changing the type of vol? It doesn't seem to be related to this patch.

Comment (singalsu) [reply src/audio/volume/volume.c]: A static analysis tool in CI nags about comparison of uint to less than zero if this is not done. There's no issues since VOL_MAX is much less than INT32_MAX.

I've not written this originally and IMHO it was a mistake to unsigned type for gain value. Negative inverting gain would be useful in some applications.

Comment (paulstelian97) [reply src/audio/volume/volume.c]: @dbaluta This reminds me of our own code failing to compile with XCC with errors related to usage of unsigned types.

@singalsu I'm not sure how negative gain even works, can you explain? I feel like the actual effect over what you hear would be nothing audible... Just destructive interference in a further mixer component, maybe?

Comment (singalsu) [reply src/audio/volume/volume.c]: Negative gain value just inverts. It could be useful with multi-way speakers and such. Some codecs/amplifiers may invert and some not. If gain can be negative then volume could align the polarity when needed. Though EQ can do it if need too.

Comment (singalsu) [reply src/audio/volume/volume.c]: @paulstelian97 Ah yes, the mixer could subtract then too.

**Comment by singalsu** [comment]
SOFCI TEST

File: src/audio/volume/volume.c (hunk 1 above)
Comment (dbaluta) [reply src/audio/volume/volume.c]: > A static analysis tool in CI nags about comparison of uint to less than zero if this is not done. There's no issues since VOL_MAX is much less than INT32_MAX.
> 
> I've not written this originally and IMHO it was a mistake to use an unsigned type for the gain value. Negative inverting gain would be useful in some applications.

This kind of explanation I would like to be present in the commit message.

Comment (singalsu) [reply src/audio/volume/volume.c]: I'll resolve this. Please let me know if not OK.

**Comment by singalsu** [comment]
Great, thanks @dbaluta , @paulstelian97 , @mmaka1 and @lgirdwood for review!
